spec/support/preload_values.rb

```diff
@@ -0,0 +1,33 @@
+# Example usage:
+#   expect { MyModel.do_the_queries }.to preload_values(:attribute, value)
+
+RSpec::Matchers.define :preload_values do |field, expected_values|
+  match(:notify_expectation_failures => true) do |block|
+    results = block.call.order(:id).load
+
+    @count = ActiveRecord::QueryCounter.count do
+      if results.respond_to?(:map)
+        expected_values = Array(expected_values) unless expected_values.kind_of?(Array)
```

Comment:
I mentioned about this line elsewhere, but if we are already checking against `:map`, doesn't that assume the use of an `Array`?

Reply:
For some tests, I found myself wanting to pass in a single value in some cases instead of an array.

I've fixed this up to work better with passing in a single value.

So the code has changed, but yes, the logic is basically the same.
(and yes, I was still not able to get the plural version working)

Reply:
Okay, yeah, I looked at this yesterday and it seemed fine.  What you have is good.